calico-vpp-agent/cni/pod_annotations.go

```diff
@@ -113,8 +118,29 @@ func (s *Server) ParseTrueFalseAnnotation(value string) (bool, error) {
 	}
 }
 
+func (s *Server) ParseSpoofAddressAnnotation(value string) ([]cnet.IPNet, error){
+	var requestedSourcePrefixes []string
+	var allowedSources []cnet.IPNet
+	err := json.Unmarshal([]byte(value), &requestedSourcePrefixes)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse '%s' as JSON: %s", value, err)
+	}
+	for _, prefix := range requestedSourcePrefixes {
+		var ipn *cnet.IPNet
+		_, ipn, err = cnet.ParseCIDROrIP(prefix)
+		if err != nil {
+			return nil, err
+		}
+		allowedSources = append(allowedSources, *(ipn.Network()))
+	}
+	return allowedSources, nil
+}
+
 func (s *Server) ParsePodAnnotations(podSpec *storage.LocalPodSpec, annotations map[string]string) (err error) {
 	for key, value := range annotations {
+		if key == CalicoAnnotationPrefix+SpoofAnnotation {
```

Comment:
Maybe this should go in the `switch key` {} ?

Reply:
we have 
```
		if !strings.HasPrefix(key, VppAnnotationPrefix) {
			continue
		}
		```
		before the switch key, so it is only for vpp anontations

Reply:
Oh, didn't see the prefix change, sorry about that